content/spin/redis-trigger.md

```diff
@@ -23,54 +30,154 @@ The Redis instance address is specified in the application trigger:
 trigger = { type = "redis", address = "redis://localhost:6379" }
 ```
 
-> We are [exploring adding authentication for connecting to Redis](https://github.com/fermyon/spin/issues/192)
-> and [adding host support for connecting to Redis databases](https://github.com/fermyon/spin/issues/181),
-> from components, which would allow using the key/value store and publishing
-> messages to channels.
+The Redis trigger also requires an `address` field.  This is the address of the Redis instance to monitor.  Specify it using the `redis:` URL scheme.
+
+> If you create an application from a Redis template, the trigger will be already set up for you.
+
+By default, Spin does not authenticate to Redis. You can work around this by providing a password in the `redis://` URL.  For example: `address = "redis://:p4ssw0rd@localhost:6379"`
+
+> We plan to offer secrets-based authentication in future versions of Spin.
+
+In addition, each component must have Redis-specific configuration in its `[component.trigger]` table.
+
+## Mapping a Channel to a Component
 
-Then, all components in the application are triggered when new messages are
-published to channels in the instance. [Configuring](./writing-apps.md) the channel
- is done by setting the `channel` field in the component trigger configuration:
+Each component handles one channel, specified in the `channel` field of the component `trigger` table.  For example, Spin will trigger this component when it receives a message on the `purchaseorders` channel:
 
 <!-- @nocpy -->
 
 ```toml
 [component.trigger]
-channel = "messages"
+channel = "purchaseorders"
+```
+
+> Spin subscribes only to the channels that are mapped to components. Other channels are ignored.
+
+## Redis Components
+
+Spin runs Redis components using the [WebAssembly component model](https://github.com/WebAssembly/component-model).  In this model, the Wasm module exports a well-known function that Spin calls to handle the Redis message.
+
+### The Message Handler
+
+The exact signature of the Redis handler, and how a function is identified to be exported as the handler, will depend on your language.
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+In Rust, the handler is identified by the `#[spin_sdk::redis_component]` attribute.  It takes a `bytes::Bytes`, representing the raw payload of the Redis message, and returns an `anyhow::Result` indicating success or an error with details.  This example just logs the payload as a string:
+
+<!-- @nocpy -->
+
+```rust
+use anyhow::Result;
+use bytes::Bytes;
+use spin_sdk::redis_component;
+use std::str::from_utf8;
+
+/// A simple Spin Redis component.
+#[redis_component]
+fn on_message(message: Bytes) -> Result<()> {
+    println!("{}", from_utf8(&message)?);
+    Ok(())
+}
+```
+
+{{ blockEnd }}
+
+{{ startTab "TypeScript"}}
+
+The JavaScript/TypeScript SDK doesn't currently support Redis components.  Please [let us know](https://github.com/fermyon/spin-js-sdk/issues) is this is important to you.
```

Comment:
```suggestion
The JavaScript/TypeScript SDK doesn't currently support Redis components.  Please [let us know](https://github.com/fermyon/spin-js-sdk/issues) if this is important to you.
```

Reply:
Oof, good catch!  Thank you for spotting that!